### PR TITLE
Introduce LoanFactory3

### PR DESCRIPTION
**Pull request by barrutko**

### CODE REVIEW

File: test/truefi2/LoanFactory3.test.ts [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        .to.be.revertedWith('LoanFactory: Loans cannot have instantaneous term of repay')
+    })
+
+    it('prevents fake pool loans', async () => {
+      const fakePool = await new TrueFiPool2__factory(owner).deploy()
+      await expect(loanFactory.connect(borrower).createLoanToken(fakePool.address, parseEth(123), 0))
+        .to.be.revertedWith('LoanFactory: Loans cannot have instantaneous term of repay')
+    })
+
+    describe('apy is set properly', () => {
```
Comment (MiksuJak) [inline test/truefi2/LoanFactory3.test.ts:112]: Maybe one more test for 2 loans at once?

File: test/truefi2/LoanFactory3.test.ts [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+  TrueFiPool2,
+  TrueFiPool2__factory,
+  TrueLender2,
+  Liquidator2,
+  PoolFactory,
+  TrueFiCreditOracle,
+  TrueFiCreditOracle__factory,
+  TrueRateAdjuster,
+  TrueRateAdjuster__factory,
+  LoanToken2,
```
Comment (skalermo) [inline test/truefi2/LoanFactory3.test.ts:18]: Can you place `LoanToken2__factory` import next to `LoanToken2`?

**Review by skalermo — APPROVED** [review]

File: contracts/truefi2/LoanFactory3.sol [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        ITrueFiPool2 _pool,
+        uint256 _amount,
+        uint256 _term
+    ) external override {
+        require(_amount > 0, "LoanFactory: Loans of amount 0, will not be approved");
+        require(_term > 0, "LoanFactory: Loans cannot have instantaneous term of repay");
+        require(poolFactory.isPool(address(_pool)), "LoanFactory: Pool was not created by PoolFactory");
+
+        uint256 apy = rate(_pool, msg.sender, _amount);
+        apy = apy.add(rateAdjuster.fixedTermLoanAdjustment(_term));
```
Comment (vanruch) [inline contracts/truefi2/LoanFactory3.sol:108]: Move this addition to `rate()`

**Review by vanruch — APPROVED** [review]

**Comment by yuchenlintt** [comment]
I manually diffed against LoanFactory2.sol, ILoanFactory2.sol, LoanFactory2.test.ts, and setupTruefi2.ts. Overall LGTM.

In the future, here's an approach we can take that breaks this kind of messy change into smaller commits. I believe there's a variant of this where we can preserve git blame history for both files:
PR 1:
  - commit that's just a git mv LoanFactory2 -> LoanFactory3
  - commit that's just a git revert deletion of LoanFactory2
PR 2:
  - commit that makes the changes we want to LoanFactory3

**Review by yuchenlintt — APPROVED** [review]

**Pull request merged**